File: Find the Duplicate Number/sol.js
// somewhat less than o(n^2)

/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    
    for(let i=0;i<nums.length;i++){
       const elem = nums[i];
        for(let j=0;j<i;j++){
            if(nums[j]==elem) return nums[j];
        }
    }

    return -1;
};

File: Find the Duplicate Number/sol.java
class Solution {
    public int findDuplicate(int[] nums) {
     
     // 0(n) space + 0(n) time 

     boolean [] present = new boolean[nums.length+1];

     for(int i=0;i<nums.length;i++){

         if(!present[nums[i]]){
             present[nums[i]] =true;
         }else{
             return nums[i];
         }
     }

     return -1;
        
    }
}